cli/src/program.rs

```diff
@@ -2730,6 +2784,22 @@ fn report_ephemeral_mnemonic(words: usize, mnemonic: bip39::Mnemonic) {
     eprintln!("[BUFFER_ACCOUNT_ADDRESS] argument to `solana program close`.\n{divider}");
 }
 
+fn set_compute_budget_ixs_if_needed(
+    ixs: &mut Vec<Instruction>,
+    compute_unit_price: &Option<u64>,
+    ixs_len: usize,
+) {
+    if let Some(compute_unit_price) = compute_unit_price {
+        println!("ixs_len: {}", ixs_len);
+        ixs.push(ComputeBudgetInstruction::set_compute_unit_limit(
+            ((ixs_len + 2) as u32) * 200_000u32,
```

Comment:
most likely this is a _drastic_ over-reservation which may reduce the likelihood of the transaction being included toward the end of a block

Reply:
i have also read that setting compute_unit_limit is optional when setting compute_unit_price wyt?

Reply:
It is optional, but it's recommended to set a limit because the default limit is 200k * the number of instructions in the transaction. The transaction priority fee is calculated by multiplying the micro-lamport fee (per compute unit) by the transaction's compute unit limit. So if you set the compute unit too high or if you don't specify one and use the default limit, you will end up overpaying priority fees.. in other words, wasting tokens.

This `set_compute_budget_ixs_if_needed` should probably accept `compute_unit_limit` argument explicitly and then each caller can calculate the exact compute unit limit needed for the passed `ixs`. Then this function can add 300 compute units to the limit for the two additional compute budget program instructions. The default compute units charged per core program instruction are as follows:
- System program: 150
- Compute budget program: 150
- Upgradeable BPF loader program: 2370

Reply:
will add this logic ty

Reply:
so we need to check each ix in ix vec and calculate compute_units required in this case right?

Reply:
Yes each ix has to be counted. I already explained this above and gave my suggestion for how to implement the fix.